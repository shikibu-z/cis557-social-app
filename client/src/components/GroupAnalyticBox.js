import React, { useState, useEffect } from 'react';
import {
  Row, Col, ListGroup, Tab,
} from 'react-bootstrap';
import '../stylesheets/GroupAnalyticBox.css';
import { getGroupAnalytic } from '../api/fetchers';

function GroupAnalyticBox(props) {
  const { groupId } = props;
  // TODO: use fetcher to get analytics
  const [Members, setMembers] = useState(1);
  const [PostDeleted, setDeleted] = useState(1);
  const [PostNumber, setNumber] = useState(1);
  const [PostHidden, setHidden] = useState(1);
  const [PostFlagged, setFlagged] = useState(1);

  useEffect(async () => {
    const analytics = await getGroupAnalytic(groupId);
    setMembers(analytics.data.memberCount);
    setDeleted(analytics.data.deleted);
    setNumber(analytics.data.postCount);
    setHidden(analytics.data.hidden);
    setFlagged(analytics.data.flagged);
  }, [Members, PostDeleted, PostNumber, PostHidden, PostFlagged]);

  return (
    <div className="AnalyticsContainer">
      <div className="AnalyticBox">
        <Tab.Container id="Analytics-tabs" defaultActiveKey="#Groups">
          <Row>
            <Col sm={4}>
              <ListGroup>
                <ListGroup.Item action href="#Groups">
                  Groups
                </ListGroup.Item>
                <ListGroup.Item action href="#Posts">
                  Posts
                </ListGroup.Item>
              </ListGroup>
            </Col>
            <Col sm={8}>
              <Tab.Content>
                <Tab.Pane eventKey="#Groups">
                  <ListGroup>
                    <ListGroup.Item>
                      Members: {Members}
                    </ListGroup.Item>
                  </ListGroup>
                </Tab.Pane>
                <Tab.Pane eventKey="#Posts">
                  <ListGroup>
                    <ListGroup.Item>
                      TotalPosts: {PostNumber}
                    </ListGroup.Item>
                    <ListGroup.Item>
                      DeletedPost: {PostDeleted}
                    </ListGroup.Item>
                    <ListGroup.Item>
                      HiddenPost: {PostHidden}
                    </ListGroup.Item>
                    <ListGroup.Item>
                      FlaggedPost: {PostFlagged}
                    </ListGroup.Item>
                  </ListGroup>
                </Tab.Pane>
              </Tab.Content>
            </Col>
          </Row>
        </Tab.Container>
      </div>
    </div>
  );
}

export default GroupAnalyticBox;
