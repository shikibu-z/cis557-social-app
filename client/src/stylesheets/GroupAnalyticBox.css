.AnalyticBox {
  margin-left: 10%;
  width: 50%;
  height: 150px;
  margin-bottom: 2%;
}