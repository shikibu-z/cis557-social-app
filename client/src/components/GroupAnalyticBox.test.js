/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import GroupAnalyticBox from './GroupAnalyticBox';

test('GroupAnalyticBox popup matches snapshot', () => {
  const component = renderer.create(<GroupAnalyticBox />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
