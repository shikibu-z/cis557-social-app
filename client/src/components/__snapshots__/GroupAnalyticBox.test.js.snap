// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupAnalyticBox popup matches snapshot 1`] = `
<div
  className="AnalyticsContainer"
>
  <div
    className="AnalyticBox"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-4"
      >
        <div
          className="list-group"
          onKeyDown={[Function]}
          role="tablist"
        >
          <a
            aria-controls="Analytics-tabs-tabpane-#Groups"
            aria-selected={true}
            className="list-group-item active list-group-item-action"
            data-rr-ui-event-key="#Groups"
            href="#Groups"
            id="Analytics-tabs-tab-#Groups"
            onClick={[Function]}
            role="tab"
          >
            Groups
          </a>
          <a
            aria-controls="Analytics-tabs-tabpane-#Posts"
            className="list-group-item list-group-item-action"
            data-rr-ui-event-key="#Posts"
            href="#Posts"
            id="Analytics-tabs-tab-#Posts"
            onClick={[Function]}
            role="tab"
            tabIndex={-1}
          >
            Posts
          </a>
        </div>
      </div>
      <div
        className="col-sm-8"
      >
        <div
          className="tab-content"
        >
          <div
            aria-labelledby="Analytics-tabs-tab-#Groups"
            className="tab-pane active"
            id="Analytics-tabs-tabpane-#Groups"
          >
            <div
              className="list-group"
              onKeyDown={[Function]}
            >
              <div
                className="list-group-item"
                data-rr-ui-event-key={null}
                onClick={[Function]}
              >
                Members: 
                1
              </div>
            </div>
          </div>
          <div
            aria-labelledby="Analytics-tabs-tab-#Posts"
            className="tab-pane"
            id="Analytics-tabs-tabpane-#Posts"
          >
            <div
              className="list-group"
              onKeyDown={[Function]}
            >
              <div
                className="list-group-item"
                data-rr-ui-event-key={null}
                onClick={[Function]}
              >
                TotalPosts: 
                1
              </div>
              <div
                className="list-group-item"
                data-rr-ui-event-key={null}
                onClick={[Function]}
              >
                DeletedPost: 
                1
              </div>
              <div
                className="list-group-item"
                data-rr-ui-event-key={null}
                onClick={[Function]}
              >
                HiddenPost: 
                1
              </div>
              <div
                className="list-group-item"
                data-rr-ui-event-key={null}
                onClick={[Function]}
              >
                FlaggedPost: 
                1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
